### PR TITLE
[FLINK-26395][docs] Correct the wrong description of RAND_INTEGER.

### PR DESCRIPTION
**Pull request by huwh**
## What is the purpose of the change
This pull request correct the wrong description of RAND_INTEGER function.

## Brief change log
Document update.

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: docs/data/sql_functions.yml [hunk 1]
```diff
@@ -221,10 +221,10 @@ arithmetic:
     description: Returns a pseudorandom double value in the range [0.0, 1.0) with an initial seed integer. Two RAND functions will return identical sequences of numbers if they have the same initial seed.
   - sql: RAND_INTEGER(INT)
     table: randInteger(INT)
-    description: Returns a pseudorandom double value in the range [0.0, INT)
+    description: Returns a pseudorandom integer value in the range [0.0, INT)
```
Comment (libenchao) [inline docs/data/sql_functions.yml:224]: `[0, INT)`

Comment (libenchao) [reply docs/data/sql_functions.yml:224]: please also update 'sql_functions_zh.yml'

**Comment by huwh** [comment]
@MartijnVisser @libenchao Thanks for the remind. I have also updated the related Chinese documentation and some code comments, please help me review these.

**Review by MartijnVisser — APPROVED** [review]
@huwh Thanks for addressing the feedback comments and fixing the PR. I've squashed commits and rebased; if the CI passes, I'll merge it. LGTM

**Comment by MartijnVisser** [comment]
@huwh After this has been merged, can you also create backports of this PR to the `release-1.15` and `release-1.14` branch?

**Comment by MartijnVisser** [comment]
@flinkbot run azure

**Comment by huwh** [comment]
@MartijnVisser Thanks for the review, I will create backports after this merged.

**Comment by MartijnVisser** [comment]
@huwh Merged it, looking forward to the backports!

**Pull request merged**